devito/types/dense.py

```diff
@@ -981,6 +987,24 @@ def space_order(self):
         """The space order."""
         return self._space_order
 
+    @property
+    def coefficients(self):
+        """Form of the coefficients of the function."""
+        return self._coefficients
+
+    @cached_property
+    def _coeff_symbol(self):
+        if self.coefficients == 'symbolic':
+            return sympy.Function('W')
+        else:
+            raise ValueError("Function was not declared with symbolic "
+                             "coefficients.")
+
+    @property
+    def coeff_symbol(self):
+        """The symbol representing the functions coefficients."""
```

Comment:
the Function coefficients ?

but more importantly, why should we expose it publicly ? isn't ``_coeff_symbol`` enough?

Reply:
We need to access this from other places within devito. This is probably my python ignorance but is it not the convention that private properties only be accessed from within the class (or directly related classes) itself?

Reply:
In general, you're right, but this class is exposed to the API and in Devito we give priority to user's visibility. "Should the user see ``coeff_symbol``?" --> no, so make it private